### PR TITLE
Fix: schedule delayed part checks correctly

### PR DESCRIPTION
**Pull request by devcrafter**
Schedule the closest delayed part check if no part checks were selected to execute. Otherwise, delayed checks won't be executed until a new check is enqueued (i.e. task is scheduled in thread pool again) 

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit f2acc366f721b5f50499c6e4d7a6762c7ac4dc4d with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56123/f2acc366f721b5f50499c6e4d7a6762c7ac4dc4d/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6693124244">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56123/f2acc366f721b5f50499c6e4d7a6762c7ac4dc4d/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Storages/MergeTree/ReplicatedMergeTreePartCheckThread.cpp [hunk 1]
```diff
@@ -555,17 +555,30 @@ void ReplicatedMergeTreePartCheckThread::run()
...
+                // find next part to check in the queue and schedule the check
+                // otherwise, scheduled for later checks won't be executed until a new check is enqueued (i.e. task is scheduled again)
+                auto next_it =
+                    std::min_element(begin(parts_queue), end(parts_queue),
+                                     [](const auto &l, const auto &r) {
+                                       return l.time < r.time;
+                                     });
+                if (next_it != parts_queue.end()) {
+                    auto delay = next_it->time - current_time;
+                    task->scheduleAfter(delay * 1000);
```
Comment (tavplubix) [inline src/Storages/MergeTree/ReplicatedMergeTreePartCheckThread.cpp:571]: Isn't it equivalent to simply checking if the queue is empty? What's the point of searching the min element?

Comment (devcrafter) [reply src/Storages/MergeTree/ReplicatedMergeTreePartCheckThread.cpp:571]: No, it isn't. The code path reaches this place if no check was selected. So, here the queue either empty or contains element(s) with delayed check. So, `min_element` will find the nearest one (by time) to check and schedule it according to its desired time to execute. If we don't schedule the delayed check here, then the time for the delayed check to execute depends on if anyone will enqueue a part check.

**Review by tavplubix — APPROVED** [review]

**Comment by tavplubix** [comment]
If test failures are not related, then you can just merge.
If they are, then restarting CI will not help.

**Comment by tavplubix** [comment]
But it still makes sense to take a look at failed tests and try to debug. For example, [Upgrade check (debug)](https://s3.amazonaws.com/clickhouse-test-reports/56123/b3b3d5ff0c174c32e868364425009909478c63a1/upgrade_check__debug_.html) has found an actual bug that leads to data corruption: https://github.com/ClickHouse/ClickHouse/issues/56202
We should never ignore crashes in stress tests.

**Pull request merged**